build-ps/build-binary.sh

```diff
@@ -443,6 +464,75 @@ fi
     ) || exit 1
     fi
 
+    # Extract the Percona Xtrabackup binaries
+    # Look for the pxb 2.4 tarball
+    (
+        cd "$TARGETDIR"
+        pxb_tar=$(ls -1td percona-xtrabackup-2.4.* | grep ".tar" | sort --version-sort | tail -n1)
+        if [[ -z $pxb_tar ]]; then
+            echo "Could not find percona-xtrabackup-2.4 tarball in $TARGETDIR.  Terminating."
+            exit 1
+        fi
+        pxb_dir=$(echo $pxb_tar | grep -oe ".*x86_64" )
+
+        mkdir -p pxc_extra
+        cd pxc_extra
+        if [[ -d ${pxb_dir} ]]; then
+            echo "Using existing pxb 2.4 directory"
+        else
+            echo "Removing existing percona-xtrabackup-2.4 basedir (if found)"
+            find . -maxdepth 1 -type d -name 'percona-xtrabackup-2.*' -exec rm -rf {} \+
+
+            echo "Extracting pxb 2.4 tarball"
+            tar -xzf "../$pxb_tar"
+        fi
+        rm -f "pxb-2.4"
+        echo "Creating symlink pxc_extra/pxb-2.4 --> $pxb_dir"
+        ln -s "$pxb_dir" pxb-2.4
+    ) || exit 1
+
+    pxb_dir=$(ls -1td $TARGETDIR/pxc_extra/percona-xtrabackup-2.* | sort --version-sort | tail -n1)
+    pxb2_version=$(echo $pxb_dir | grep -oe "[1-9]\.[0-9][0-9]*\.[0-9][0-9]*")
+
+    # Look for the pxb 8.0 tarball
+    (
+        cd "$TARGETDIR"
+        pxb_tar=$(ls -1td percona-xtrabackup-8.0.* | grep ".tar" | sort --version-sort | tail -n1)
+        if [[ -z $pxb_tar ]]; then
+            echo "Could not find percona-xtrabackup-8.0.x tarball in $TARGETDIR.  Terminating."
+            exit 1
+        fi
+        pxb_dir=$(echo $pxb_tar | grep -oe ".*x86_64" )
+
+        mkdir -p pxc_extra
+        cd pxc_extra
+        if [[ -d ${pxb_dir} ]]; then
+            echo "Using existing pxb 8.0 directory"
+        else
+            echo "Removing existing percona-xtrabackup-8.0 basedir (if found)"
+            find . -maxdepth 1 -type d -name 'percona-xtrabackup-8.*' -exec rm -rf {} \+
+
+            echo "Extracting pxb 8.0 tarball"
+            tar -xzf "../$pxb_tar"
+        fi
+        rm -f "pxb-8.0"
+        echo "Creating symlink pxc_extra/pxb-8.0 --> $pxb_dir"
+        ln -s "$pxb_dir" pxb-8.0
+    ) || exit 1
+
+    pxb_dir=$(ls -1td $TARGETDIR/pxc_extra/percona-xtrabackup-8.* | sort --version-sort | tail -n1)
+    pxb8_version=$(echo $pxb_dir | grep -oe "[1-9]\.[0-9][0-9]*\.[0-9][0-9]*")
+
+    mkdir -p "$TARGETDIR/usr/local/$PRODUCT_FULL_NAME/bin/pxc_extra/pxb-2.4"
+    (cp -v -r $TARGETDIR/pxc_extra/pxb-2.4/bin/  $TARGETDIR/usr/local/$PRODUCT_FULL_NAME/bin/pxc_extra/pxb-2.4) || true
```

Comment:
Leave a comment that we explicitly copy only binary and lib. (test cases and other files are not copied).

Reply:
added a comment